src/tbb/thread_request_serializer.cpp

```diff
@@ -37,16 +37,16 @@ void thread_request_serializer::update(int delta) {
     if (prev_pending_delta == pending_delta_base) {
         delta = int(my_pending_delta.exchange(pending_delta_base) & delta_mask) - int(pending_delta_base);
         mutex_type::scoped_lock lock(my_mutex);
-        my_total_request += delta;
-        delta = limit_delta(delta, my_soft_limit, my_total_request);
+        my_total_request.store(my_total_request.load(std::memory_order_relaxed) + delta, std::memory_order_release);
+        delta = limit_delta(delta, my_soft_limit, my_total_request.load(std::memory_order_relaxed));
         my_thread_dispatcher.adjust_job_count_estimate(delta);
     }
 }
 
 void thread_request_serializer::set_active_num_workers(int soft_limit) {
     mutex_type::scoped_lock lock(my_mutex);
     int delta = soft_limit - my_soft_limit;
-    delta = limit_delta(delta, my_total_request, soft_limit);
+    delta = limit_delta(delta, my_total_request.load(std::memory_order_relaxed), soft_limit);
```

Comment:
Happens-before relationships are not built on top of `my_total_request`, thus all the operations should be `std::memory_order_relaxed`.

Reply:
Changed.